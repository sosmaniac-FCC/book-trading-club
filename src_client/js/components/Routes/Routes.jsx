import React from 'react';
import { Switch, Redirect, Route, withRouter } from 'react-router-dom';

import { connect } from 'react-redux';

import BookSpread from '../BookSpread/BookSpread.jsx';
import CreateAccount from '../CreateAccount/CreateAccount.jsx';
import Login from '../Login/Login.jsx';
import Main from '../Main/Main.jsx';
import MyBooks from '../MyBooks/MyBooks.jsx';
import NewBook from '../NewBook/NewBook.jsx';
import NewTrade from '../NewTrade/NewTrade.jsx';
import NotFound from '../NotFound/NotFound.jsx';
import Settings from '../Settings/Settings.jsx';
import Trades from '../Trades/Trades.jsx';
import Trade from '../Trade/Trade.jsx';

class Routes extends React.Component {
    constructor(props) {
        super(props);
    }
    
    protectedRoute(props, Component) {
        if (this.props.isAuthenticated) {
            return <Component {...props}/>;
        }
        else {
            return <Redirect to={{
                pathname: "/Login",
                state: "You must login first!"
            }}/>;
        }
    }
    
    render() {
        return (
            <Switch>
                <Route exact path="/"                           component={Main}/>
                <Route exact path="/Login/CreateAccount"        component={CreateAccount}/>
                <Route exact path="/Login"                      component={Login}/>
                <Route exact path="/User/:user/Settings"        render={(props) => this.protectedRoute(props, Settings)}/>
                <Route exact path="/User/:user/Trades"          render={(props) => this.protectedRoute(props, Trades)}/>      
                <Route exact path="/User/:user/Trades/New"      render={(props) => this.protectedRoute(props, NewTrade)}/>  
                <Route exact path="/User/:user/Books"           render={(props) => this.protectedRoute(props, MyBooks)}/>           
                <Route exact path="/User/:user/Books/New"       render={(props) => this.protectedRoute(props, NewBook)}/>           
                <Route exact path="/Trade/:id"                  render={(props) => this.protectedRoute(props, Trade)}/> 
                <Route exact path="/Book/BookSpread"            render={(props) => this.protectedRoute(props, BookSpread)}/>
                <Route component={NotFound}/>
            </Switch>
        );
    }
}

function mapStateToProps({ users }) {
    return {
        isAuthenticated: users.isAuthenticated
    };
}

export default withRouter(connect(mapStateToProps, null)(Routes));

// TO-DO : ACCESS CONTROL FOR PORTS USER, TRADE, & BOOK

// FOR AUTHENTICATION, USE REACT-ROUTER RENDER
// render={(props) => (<Component { ...props }/>)}