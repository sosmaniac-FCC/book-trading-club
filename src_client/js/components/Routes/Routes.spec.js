// expect from jest_expect.js
/* global expect */

import React from 'react';
import { shallow, configure } from 'enzyme';
import enzymeAdapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

import Routes from './Routes.jsx';

configure({ adapter: new enzymeAdapter() });

test('Switch component container holding all applicable routes loads as expected.', () => {
    const component = shallow(<Routes />);
    const tree = toJSON(component);
    expect(tree).toMatchSnapshot();
});

test('Switch component container holding all applicable routes should have no style attributes.', () => {
    const component = shallow(<Routes />);
    expect(component.find('[style]')).toHaveLength(0);
});