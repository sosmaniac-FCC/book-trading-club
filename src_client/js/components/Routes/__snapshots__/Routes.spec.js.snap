// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component container holding all applicable routes loads as expected. 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/Login"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/Login/CreateAccount"
  />
  <Route
    component={null}
    exact={true}
    path="/User/:user/Profile"
  />
  <Route
    component={null}
    exact={true}
    path="/User/:user/Trades"
  />
  <Route
    component={null}
    exact={true}
    path="/User/:user/Books"
  />
  <Route
    component={null}
    exact={true}
    path="/Trade/:id/Out"
  />
  <Route
    component={null}
    exact={true}
    path="/Trade/:id/In"
  />
  <Route
    component={null}
    exact={true}
    path="/Book/:id"
  />
  <Route
    component={null}
    exact={true}
    path="/Book/BookSpread"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
